homeassistant/components/stt/__init__.py

```diff
@@ -125,12 +127,22 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     )
 
     component.register_shutdown()
+
+    hass.http.register_view(SpeechToTextView(hass.data[DATA_PROVIDERS]))
+
+    # We need to add the component here break the deadlock
+    # when setting up integrations from config entries as
+    # they would otherwise wait for the stt to be
+    # setup and thus the config entries would not be able to
+    # setup their platforms.
+    hass.config.components.add(DOMAIN)
```

Comment:
This was a bug that stt was missing this (`device_tracker` and `notify` have the same pattern)